lib/mpl_toolkits/mplot3d/proj3d.py

```diff
@@ -17,11 +17,13 @@ def _line2d_seg_dist(p1, p2, p0):
     and intersection point lies within segment if u is between 0 and 1
     """
 
-    x21 = p2[0] - p1[0]
-    y21 = p2[1] - p1[1]
     x01 = np.asarray(p0[0]) - p1[0]
     y01 = np.asarray(p0[1]) - p1[1]
+    if p1 == p2:  # Avoid RuntimeWarning
```

Comment:
```suggestion
    if np.all(p1 == p2):  # Avoid RuntimeWarning
```
This will fail if p1 and p2 are numpy arrays.

```
In [3]: p1 = np.array([1, 2])

In [4]: p1 == p1
Out[4]: array([ True,  True])

In [5]: bool(p1 == p1)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Input In [5], in <cell line: 1>()
----> 1 bool(p1 == p1)


```